src/mame/capcom/lwings.cpp

```diff
@@ -1062,15 +932,83 @@ void lwings_state::trojan(machine_config &config)
 	m_msm->add_route(ALL_OUTPUTS, "mono", 0.50);
 }
 
+u8 lwings_state::mcu_p0_r()
+{
+	u8 retval = m_mcu_prot_val;
+	//logerror("%s mcu_p0_r (read protection param %02x)\n", machine().describe_context().c_str(), retval);
+	return retval;
+}
+
+u8 lwings_state::mcu_p1_r()
+{
+	// this is used to decide if we're sending angle params or a sound write? compares against 0xf0, vpos like 1943?
+	return m_screen->vpos();
+}
+
+u8 lwings_state::mcu_p2_r()
+{
+	return m_palette_pen_raw;
+}
+
+u8 lwings_state::mcu_p3_r()
+{
+	return 0x00;
+}
+
+void lwings_state::mcu_p0_w(u8 data)
+{
+	m_mcu_outlatch = data;
+}
+
+void lwings_state::mcu_p1_w(u8 data)
+{
+}
+
+void lwings_state::mcu_p2_w(u8 data)
+{
+	m_mcu->set_input_line(0, CLEAR_LINE); // not sure, we also clear on a timed ack
+	m_soundstate = 0x80;
+	m_soundlatch->write(data);
+}
```

Comment:
The I/O ports aren’t strobed, so clearing the interrupt on any write isn’t possible.